app/services/doc_auth/error_generator.rb

```diff
@@ -189,14 +200,42 @@ def get_error_messages(liveness_enabled, response_info)
         end
       end
 
-      portrait_match_results = response_info[:portrait_match_results] || {}
-      if liveness_enabled && portrait_match_results.dig(:FaceMatchResult) != 'Pass'
-        errors[SELFIE] << Errors::SELFIE_FAILURE
+      selfie_error = get_selfie_error(liveness_enabled, response_info)
+      if liveness_enabled && selfie_error
+        errors[SELFIE] << selfie_error
       end
 
       errors
     end
 
+    def get_selfie_error(liveness_enabled, response_info)
+      # The part of the response that contains information about the selfie
+      portrait_match_results = response_info[:portrait_match_results] || {}
+      # The overall result of the selfie, 'Pass' or 'Fail'
+      face_match_result = portrait_match_results.dig(:FaceMatchResult)
+      # The reason for failure (if it failed), also sometimes contains success info
+      face_match_error = portrait_match_results.dig(:FaceErrorMessage)
+
+      # No error if liveness is not enabled or if there's no failure
+      if !liveness_enabled || face_match_result == 'Pass'
+        return nil
+      end
+
+      # Error when the image on the id does not match the selfie image, but the image was acceptable
+      if face_match_result == 'Fail' &&
```

Comment:
maybe we should move to bottom? Then we only need to check _face_match_result == 'Fail'_

Reply:
I'm not sure I'm following, move which part to the bottom? This `if` statement?